### DISCUSSION TITLE
XML with XML Schema and Schematron as format options in Goodtables wished for

### DISCUSSION THREAD

**Discussion by jomtov** [frictionlessdata/frictionlessdata.io#595] [Ideas]
Motivation: Although .JSON can reasonably easily converted into XML, JSON schema is not as developed and advanced yet as XML Schema and in particular Schematron (an ISO-standard), which can be more specific and  tailormade error-messages, that are easier understandable to humans.

**Comment by rufuspollock** [comment]
@jomtov great question. Can you give more links and details and a simple example of how this would work on the command line or in code?

**Comment by jomtov** [comment]
Hi @rufuspollock,

I am not really a developer myself, so cannot give you code or command line details. But I have written some Schematron schemas, using phases also. Attaching a schema with examples, used for validation of OAI-PMH METS-feeds for output from Swedish Digital Scientific Publications  Repository – DIVA.
(First validate in Schematron phase “Diagnosis”, then choose next validation phase after result from there. Please note that the schema and examples are from 2016 and should be updated. Also, the Schematron schema is designed specifically to validate things that are NOT validated by the default ordinary XML Schema, thus leaving out e.g. the systematic ID-errors in these feeds)

Anyway, there are Schematron resources on GitHub too, such as here:
https://github.com/Schematron/schematron

Best,

@jomtov / joakim.philipson@su.se


Från: Rufus Pollock <notifications@github.com>
Skickat: den 27 juli 2020 14:54
Till: frictionlessdata/forum <forum@noreply.github.com>
Kopia: Joakim Philipson <joakim.philipson@su.se>; Mention <mention@noreply.github.com>
Ämne: Re: [frictionlessdata/forum] XML with XML Schema and Schematron as format options in Goodtables wished for (#64)


@jomtov<https://github.com/jomtov> great question. Can you give more links and details and a simple example of how this would work on the command line or in code?
